### PR TITLE
Add KEP-184 'scheduler simulation'

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind feature
/kind documentation

#### What this PR does / why we need it:

Add KEP-184 'scheduler simulation'

#### Which issue(s) this PR fixes:

Part of #184

#### Special notes for your reviewer:

This feature is built over the features introduced in KEP-140 and KEP-167. So, this PR needs to be merged after them.

Note that this KEP may need to be significantly changed if there are significant changes from what is currently proposed in KEP-140 and KEP-167.

https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/185
https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/186

/label tide/merge-method-squash

### CODE REVIEW

**Comment by sanposhiho** [comment]
> This feature is built over the features introduced in KEP-140 and KEP-167. So, this PR needs to be merged after them.
Note that this KEP may need to be significantly changed if there are significant changes from what is currently proposed in KEP-140 and KEP-167.

To avoid accidental merging. 

/hold

File: keps/184-scheduler-simulation/README.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  // Cannot be updated.
+  ScenarioResultFilePath string
+  // Volume to mount into the filesystem.
+  //
+  // The container named "scenario-runner" will be created in the simulator Pod.
+  // That container will see the scenario defined in ScenarioTemplateFilePath, create the scenario in the simulator, and store the result into ScenarioReusltFilePath.
+  // The volume specified in Volume and VolumeMount is mounted in that container.
+  //
+  // Cannot be updated.
+  VolumeMount corev1.VolumeMount
```
Comment (everpeace) [inline keps/184-scheduler-simulation/README.md:83]: Do we need to allow users to configure the path/volumeMount for them?? I suppose the fixed path/volumeMount would be enough, i.e., this raises any inconvenience for them. WDYT?

**Review by everpeace — COMMENTED** [review]
Thanks for writing up!!  I left a few comments.  This looks good to me.  

Although I think the kep stays looks good to me when many changes happened in dependents keps, let's see what's going on them　for now.

File: keps/184-scheduler-simulation/README.md (hunk 1 above)
Comment (sanposhiho) [reply keps/184-scheduler-simulation/README.md:83]: > I suppose the fixed path/volumeMount would be enough, i.e., this raises any inconvenience for them. WDYT?

I may not have understood exactly what you meant. So, you wanna remove `VolumeMount` and make `Volume` always be mounted in the same `mountPath` in "scenario-runner" container?

File: keps/184-scheduler-simulation/kep.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  - "@196Ikuchil"
+
+stage: alpha
+
+milestone:
+  alpha: "v0.1"
+
+see-also:
+  - "/keps/140-scenario-based-simulation"
+  - "/keps/167-scheduler-simulator-operator"
```
Comment (196Ikuchil) [inline keps/184-scheduler-simulation/kep.yaml:21]:
```suggestion
  - "/keps/159-scheduler-simulator-operator"
```

File: keps/184-scheduler-simulation/README.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+```
+
+#### SchedulerSimulation execution flow
+
+0. User creates SchedulerSimulation resource in user's cluster.
+1. The SchedulerSimulation controller creates a Simulator resource with "scenario-runner" container.
+2. After the Pod gets started, the "scenario-runner" reads ScenarioTemplateFile and creates the Scenario resource in simulator's kube-apiserver.
+3. The simulator run the scenario.
+4. After the scenario is done, the "scenario-runner" fetch the Scenario's result from simulator's kube-apiserver and store it in ScenarioResultFile.
+5. The "scenario-runner" container is terminated.
```
[the remark this answers is not in the document]
Comment (sanposhiho) [reply keps/184-scheduler-simulation/README.md:159]: Nice question. All your guessing is correct.

Users can define the scenario that doesn't have "Done" operation. And, in that case, the user can access the simulator to add the more scenarios by looking at the simulation results and the status of the resource at that point.

File: keps/184-scheduler-simulation/kep.yaml (hunk 2 above)
Comment (sanposhiho) [reply keps/184-scheduler-simulation/kep.yaml:21]: nice catch. will fix later

File: keps/184-scheduler-simulation/README.md (hunk 1 above)
Comment (sanposhiho) [reply keps/184-scheduler-simulation/README.md:83]: make sense.

**Comment by sanposhiho** [comment]
@everpeace @196Ikuchil Updated as your reviews.

**Review by everpeace — APPROVED** [review]
LGTM!

*I put off `/lgtm` not to get this PR merged because other reviewers are still in review.*

**Comment by 196Ikuchil** [comment]
Thanks!
This feature would make the simulator much more useful!

/lgtm

**Pull request merged**